### PR TITLE
feat(debug): expose playwright object in console

### PR DESCRIPTION
**Pull request by dgozman**
Under `PLAYWRIGHT_DEBUG_UI`:
- `playwright.$(selector)` and `playwright.$$(selector)` to query elements;
- `playwright.inspect(selector)` to reveal an element;
- `playwright.clear()` to remove highlight.

### CODE REVIEW

File: src/chromium/crBrowser.ts [hunk 1]
```diff
@@ -301,6 +302,7 @@ export class CRBrowserContext extends BrowserContextBase {
     }
     if (this._options.permissions)
       promises.push(this.grantPermissions(this._options.permissions));
+    promises.push(debugSupport.installConsoleHelpers(this));
```
Comment (pavelfeldman) [inline src/chromium/crBrowser.ts:305]: Can we have this in a browser-neutral manner?

File: src/debug/debugSupport.ts [hunk 2]
```diff
@@ -45,3 +51,86 @@ export async function generateSourceMapUrl(functionText: string, generatedText:
...
+  const installInFrame = async (frame: Frame) => {
+    try {
+      const main = await frame._mainContext();
+      const injectedScript = await main.injectedScript();
+      await injectedScript.evaluate((injectedScript, parseSelectorString) => {
+        const parseSelector: (selector: string) => types.ParsedSelector = new Function('...args', 'return (' + parseSelectorString + ')(...args)') as any;
+        const checkSelector = (parsed: types.ParsedSelector) => {
+          for (const {name} of parsed.parts) {
+            if (!injectedScript.engines.has(name))
+              throw new Error(`Unknown engine "${name}"`);
```
Comment (pavelfeldman) [inline src/debug/debugSupport.ts:67]: Can we do something about the nesting level here? I can't figure out where parseSelectorString is coming from.

File: src/debug/debugSupport.ts [hunk 3]
```diff
@@ -45,3 +51,86 @@ export async function generateSourceMapUrl(functionText: string, generatedText:
...
+          }
+        };
+
+        const glassPane = document.createElement('div');
+        glassPane.style.cssText = 'position: absolute; left: 0; top: 0; pointer-events: none; overflow: visible; z-index: 10000;';
```
Comment (pavelfeldman) [inline src/debug/debugSupport.ts:72]: How do you use this glass pane?

File: src/debug/debugSupport.ts [hunk 4]
```diff
@@ -16,6 +16,12 @@
 
 import * as sourceMap from './sourceMap';
 import { getFromENV } from '../helper';
+import { BrowserContextBase } from '../browserContext';
+import { Frame } from '../frames';
+import { Events } from '../events';
+import { Page } from '../page';
+import { parseSelector } from '../selectors';
```
Comment (pavelfeldman) [inline src/debug/debugSupport.ts:23]: Doesn't injected script know how to parse selectors?

File: src/chromium/crBrowser.ts (hunk 1 above)
Comment (dgozman) [reply src/chromium/crBrowser.ts]: Wrapped in `BrowserContextBase._initialize`.

File: src/debug/debugSupport.ts (hunk 4 above)
Comment (dgozman) [reply src/debug/debugSupport.ts:23]: Nope, we parse them in Node js to handle main/utility worlds and to provide better error messages.

File: src/debug/debugSupport.ts (hunk 3 above)
Comment (dgozman) [reply src/debug/debugSupport.ts]: It is zero-by-zero at the top-left, only used for positioning the highlights. Let me rename it to container.

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**